core/templates/services/user.service.ts

```diff
@@ -56,21 +60,28 @@ export class UserService {
     return this.userInfo;
   }
 
-  async getProfileImageDataUrlAsync(): Promise<string> {
-    let defaultUrl = (
-      this.urlInterpolationService.getStaticImageUrl(
-        AppConstants.DEFAULT_PROFILE_IMAGE_PATH));
-    return this.getUserInfoAsync().then(
-      async(userInfo) => {
-        if (userInfo.isLoggedIn()) {
-          return this.userBackendApiService.getProfileImageDataUrlAsync(
-            defaultUrl);
-        } else {
-          return new Promise((resolve, reject) => {
-            resolve(defaultUrl);
-          });
-        }
-      });
+  getProfileImageDataUrlAsync(username: string): [string, string] {
+    let defaultUrlWebp = this.urlInterpolationService.getStaticImageUrl(
+      AppConstants.DEFAULT_PROFILE_IMAGE_WEBP_PATH);
+    let defaultUrlPng = this.urlInterpolationService.getStaticImageUrl(
+      AppConstants.DEFAULT_PROFILE_IMAGE_PNG_PATH);
+    let localStoredImage = this.imageLocalStorageService.getRawImageData(
+      username + '_profile_picture.png');
+    if (localStoredImage !== null) {
+      return [localStoredImage, localStoredImage];
```

Comment:
Please add an explanatory comment about this.

Reply:
Done